### ISSUE TITLE
Changing black clamp level directly in hmd config

### ISSUE THREAD

**Issue by QUAKEULUS** [ValveSoftware/openvr#1286]
I'm trying to set the black levels directly in the HMD instead of relying on debug tools, I understand this can have a side effect of black smearing. 

I have only HMD connected, I run lighthouse_console.exe in  \steamapps\common\SteamVR\tools\lighthouse\bin\win32

I download the config, 

line 30:         "black_clamp":  0.0117647061124444
I tried changing it to another value, but there seems to be no effect.
As I understand it's set to 3, because  0.0117647061124444*255 is 3
I tried 0.0 and 1 (0.00392156862) and even 10 (0.03921568627), no difference in black levels

Is that not the same as black_clamp debug console command that was in steamvr a few versions ago?

Please advise

**Comment by haywirephoenix** [comment]
Please re enable black clamp and mura correction toggles in the debug commands and a black clamp level slider

**Comment by haywirephoenix** [comment]
Incase anyone wants the last known steamvr build with black_clamp and mura_correction toggles here's a link to 1.5.3 :

https://drive.google.com/open?id=1-O58ETo7a0Rbf9aYHpYUQMrSaskUweiY

Note you may have an input issue with some games.

**Comment by QUAKEULUS** [comment]
That is one issue. I made a similar package of old steamVR that has the debug command here:
https://www.reddit.com/r/vive_vr/comments/eeym47/hey_guys_i_made_an_easier_package_for_older/

But I tried to circumvent that and apply black_clamp levels directly into the hmd, so I would not face this issue, and perhaps have it not be completely clamped off, but I wrote the values to the HMD into its config but it seems to be ignoring that as well.

**Comment by haywirephoenix** [comment]
> That is one issue. I made a similar package of old steamVR that has the debug command here:
> https://www.reddit.com/r/vive_vr/comments/eeym47/hey_guys_i_made_an_easier_package_for_older/
> 
> But I tried to circumvent that and apply black_clamp levels directly into the hmd, so I would not face this issue, and perhaps have it not be completely clamped off, but I wrote the values to the HMD into its config but it seems to be ignoring that as well.
> 
> It would be nice to have 1. the debug command for those that want to temporarily use it
> 2. Have the values written to the HMD be used for a more permanent solution.

I tried changing those values before too. It's funny how they removed true black just before the index was released.

**Comment by pavel-ruban** [comment]
Have Vive Pro HMD, black levels look awful, when I check with non direct mode the blue with pitch black settings it's ideal & for dark games I have no issues with smearing on it, it's noticable but for me it's minor, but that grey black on so good oled HMD kills,

I tried to upload config file with black_clamp equal to 0.0 by lighthouose_console.exe as well - no effect.

Please reenable mura & black clamp options in debug tool, or even better to do such options for oled screens in steamvr UI

**Comment by QUAKEULUS** [comment]
@pavel-ruban 
Same issue. Mura is a separate issue from black clamp, and it makes the lack of blacks look worse.
Are you wireless? Because there seems to be an additional issue using htc wireless with mura correction: https://www.reddit.com/r/Vive/comments/9n9k8i/does_wireless_adapter_break_mura_correction_on/

I also have a samsung odyssey plus, which afaik has identical panels as vive pro, and it has both the best optics when it comes to decreased godrays, and very faint black smear. It's not black or purple like on other HMDs, it's a light grey smear.

I really wish mura correction and black smear would be re-implemented. I don't understand why those parameters are ignored while the distortion profiles are not in the config.

**Comment by pavel-ruban** [comment]
I have many HDMs & Samsung O+ as well, the black smearing on it is awefull though, so it makes sense why black is clamped by default, but it's not the case for all the games, in some games where you don't shake or turn your head really quickly the lack of pitch black is terrible.

Once I first time connected my Vive Pro HMD I had really noticeable mura, but blacks were ok, then it asked for update & mura correction did the great job but update also killed the blacks.

I mentioned Mura option  just to have additional option for it. I actually have zero complains about mura but I don't get why people can't disablei it, as many complain about different mura profiles for different eyeys & it causes headache for HDMs where it wasn't configured properly. I don't use wireless.

Regarding godrays - the answer is gear vr mod, apart of godrays native pro lenses have so bad sweetspot so turning your head makes image kind of smear, but it's not a smear actually its lenses problems as things aren't in focus & they look kind of smeared, gear vr mod increases sweetspot drastically & visuals become much better.

**Comment by haywirephoenix** [comment]
I never noticed smearing on any hmd before it was clamped. It seems to bother some more than others,o why remove our ability to choose. (Right before the Valve Index release btw which is LCD). I notice it when looking for it and scrolling on my oled smartphone but only when the brightness is at it's lowest and it's light image scrolling against dark. Maybe the brightness/refresh rate can reduce the smear.

Some game's gamma levels are worse than others, eg Arizona Sunshine (where darkness is supposed to be part of the experience) for me the blacks are pale grey whereas much deeper on HL:A. I got the last black_clamp steamvr build working on Arizona Sunshine but it didnt make a huge difference and showed the mura, maybe this is why they removed it. We have no colour controls at all so black_clamp is the only chance we had. The new Pimax PiTool update has colour controls but also started clamping black aswell with no disable option. 

@QUAKEULUS is O+ worth getting, how is WMR? 
@pavel-ruban Did you manage to find a perfect distortion profile post GearVR mod? I heard there isn't a perfect one and people get sickness. My gearvr lenses have noticeable warping for sure.
Vive has the worst godrays and this is why https://cdn.mos.cms.futurecdn.net/4SCFndUPt7EB9nja32CGWH-1200-80.jpg

Return/sell your Vive pro's guys while you still can. It's not worth the money imo.

**Comment by haywirephoenix** [comment]
@pavel-ruban Fair enough, I had some trouble getting non direct mode to play nicely but remember it not being perfect. Interesting you prefer the VivePros displays, they use the exact same displays as in the O+ except they put a blurry sticker on theirs doe the Anti-SDE. Maybe it can be peeled off of it's not sharp enough? Those were going for ~$250 at one point, for the same displays, I heard the controller tracking wasnt great though, you could maybe mix it with knuckles and vive trackers.

**Comment by pavel-ruban** [comment]
XR isn't good at all, it uses the same panels as O+ & Vive Pro but firmwarewise it's nog optimized, dull & dimm + it has much worse SDE as image is stretched across big FOV, it's way better than on OG Vive but far beyond the Vive Pro.
I didn't have problems with latest pitool & there is option pitch black, it works, black is pure black. Colours are ok but they locked in firmware & there is not enough brightness, I guess it can be enhanced. FOV although is great, but I 5k+ is much better option than 5kXR until they manage proper panels utilization & adjust brightness, find XR short review here https://community.openmr.ai/t/5kxr-5k-impressions/28669

O+ is good HMD, but tracking is bad, you can't play bow games, throw granedes normally. It works decent when system sees controllers but there are only 2 cameras in front of you. Although for FPS & all other kind of games like beat saber it's ok. But I don't like controllers & they drain battareys (2 per one controller) too fast (4-5 hours)

I know you can mix drivers, but afaik there are some caveats & probably to mitigate some you need to attach vive tracker to your HMD, so I didn't bother.

Display on O+ really good but I prefer Vive Pro in this regard. On O+ it's blurry. Some ppl preffer OG O coz it hadn't this glues film.

For me personally Vivo Pro provides better immersion, you feel 3d depth.
Colour & Brightness are really good on O+, but I don't like WMR soft (it's ok though) + it suffers from black smear a lot without any exposed settings to fix it, although I saw some posts where people got some good results with some hacks.

Also for me subjectevly comfort of O+ is poor. It has great audio though.

Vive Pro is comfortable & I use it with Valve Knuckles, I also don't like wands I bought HMD alone as I already had base stations & knuckles, HMD costed about 800$

**Comment by deorder** [comment]
Does someone still have the version with those options in it?

**Comment by QUAKEULUS** [comment]
> 
> 
> Does someone still have the SteamVR version with those options in it?

https://1fichier.com/?42fk1sbw4cnovqjnzs4o

**Comment by Cless-Aurion** [comment]
Okay so. I did try clamping my blacks through the Nvidia panel some time ago on my OG-Vive. Black smear was NOTICEABLE, and really, the blacks weren't worth it THAT much (even if I'm obsessed with them), so that and the hassle of having direct mode I just reverted back.

Today I gave a try to this old version of SteamVR on my Vive Pro to "clamp blacks"... And are you telling me, the ONLY downside, is a SLIGHTLY BARELY noticeable purple tint on bright borders only when you move fast your head!?

What-the-actual-fuck.

HOW IS THIS NOT THE DEFAULT FOR THE PRO!?

Christ, this is SO frustrating now, knowing that the headset is perfectly capable, even if with a minor barely noticeable inconvenience.
